From a33c041c8d2c0596719f4c22277f176c5e8b8dd8 Mon Sep 17 00:00:00 2001
Message-Id: <patch-1.1-a33c041c8d2-20220615T103046Z-avarab@gmail.com>
In-Reply-To: <cover-0.1-00000000000-20220615T103046Z-avarab@gmail.com>
References: <cover-0.1-00000000000-20220615T103046Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Tue, 14 Jun 2022 19:07:42 +0200
Subject: [PATCH 1/1] tests: add CURL prerequisite to tests needing libcurl
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add and use a CURL prerequisite for tests added in
6dcbdc0d661 (remote: create fetch.credentialsInUrl config,
2022-06-06).

These tests would get as far as emitting a couple of the warnings we
were testing for, but would then die as we had no "git-remote-https"
program compiled.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 t/lib-httpd.sh        | 2 +-
 t/t5516-fetch-push.sh | 4 ++--
 t/t5601-clone.sh      | 4 ++--
 t/test-lib.sh         | 1 +
 4 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/t/lib-httpd.sh b/t/lib-httpd.sh
index 782891908d7..878daadfbd9 100644
--- a/t/lib-httpd.sh
+++ b/t/lib-httpd.sh
@@ -29,7 +29,7 @@
 # Copyright (c) 2008 Clemens Buchacher <drizzd@aon.at>
 #
 
-if test -n "$NO_CURL"
+if ! test_have_prereq CURL
 then
 	skip_all='skipping test, git built without http support'
 	test_done
diff --git a/t/t5516-fetch-push.sh b/t/t5516-fetch-push.sh
index dedca106a7a..c69e252ae7f 100755
--- a/t/t5516-fetch-push.sh
+++ b/t/t5516-fetch-push.sh
@@ -1834,7 +1834,7 @@ test_expect_success 'refuse to push a hidden ref, and make sure do not pollute t
 	test_dir_is_empty testrepo/.git/objects/pack
 '
 
-test_expect_success 'fetch warns or fails when using username:password' '
+test_expect_success CURL 'fetch warns or fails when using username:password' '
 	message="URL '\''https://username:<redacted>@localhost/'\'' uses plaintext credentials" &&
 	test_must_fail git -c fetch.credentialsInUrl=allow fetch https://username:password@localhost 2>err &&
 	! grep "$message" err &&
@@ -1853,7 +1853,7 @@ test_expect_success 'fetch warns or fails when using username:password' '
 '
 
 
-test_expect_success 'push warns or fails when using username:password' '
+test_expect_success CURL 'push warns or fails when using username:password' '
 	message="URL '\''https://username:<redacted>@localhost/'\'' uses plaintext credentials" &&
 	test_must_fail git -c fetch.credentialsInUrl=allow push https://username:password@localhost 2>err &&
 	! grep "$message" err &&
diff --git a/t/t5601-clone.sh b/t/t5601-clone.sh
index d2f046b4b92..bfbaa0debac 100755
--- a/t/t5601-clone.sh
+++ b/t/t5601-clone.sh
@@ -71,7 +71,7 @@ test_expect_success 'clone respects GIT_WORK_TREE' '
 
 '
 
-test_expect_success 'clone warns or fails when using username:password' '
+test_expect_success CURL 'clone warns or fails when using username:password' '
 	message="URL '\''https://username:<redacted>@localhost/'\'' uses plaintext credentials" &&
 	test_must_fail git -c fetch.credentialsInUrl=allow clone https://username:password@localhost attempt1 2>err &&
 	! grep "$message" err &&
@@ -89,7 +89,7 @@ test_expect_success 'clone warns or fails when using username:password' '
 	test_line_count = 1 warnings
 '
 
-test_expect_success 'clone does not detect username:password when it is https://username@domain:port/' '
+test_expect_success CURL 'clone does not detect username:password when it is https://username@domain:port/' '
 	test_must_fail git -c fetch.credentialsInUrl=warn clone https://username@localhost:8080 attempt3 2>err &&
 	! grep "uses plaintext credentials" err
 '
diff --git a/t/test-lib.sh b/t/test-lib.sh
index 736c6447ecf..9afad04e215 100644
--- a/t/test-lib.sh
+++ b/t/test-lib.sh
@@ -1572,6 +1572,7 @@ esac
 test_set_prereq REFFILES
 
 ( COLUMNS=1 && test $COLUMNS = 1 ) && test_set_prereq COLUMNS_CAN_BE_1
+test -z "$NO_CURL" && test_set_prereq CURL
 test -z "$NO_PERL" && test_set_prereq PERL
 test -z "$NO_PTHREADS" && test_set_prereq PTHREADS
 test -z "$NO_PYTHON" && test_set_prereq PYTHON
-- 
2.36.1.1239.gfba91521d90

